x-pack/plugins/enterprise_search/common/types/app_search.ts

```diff
@@ -23,3 +23,42 @@ export interface IRole {
     availableRoleTypes: string[];
   };
 }
+
+export interface IApiToken {
+  access_all_engines?: boolean;
+  key?: string;
+  engines?: string[];
+  id?: number;
+  name: string;
+  read?: boolean;
+  type: string;
+  write?: boolean;
+}
+
+export interface IMetaPage {
+  current: number;
+  size: number;
+  total_pages: number;
+  total_results: number;
+}
+
+export interface IMeta {
+  page: IMetaPage;
+}
+
+export interface IEngine {
+  name: string;
+  type: string;
+  language: string;
+  result_fields: object[];
+}
+
```

Comment:
These are shared and I have a local branch that I am using them in and will be putting them in [this file](https://github.com/elastic/kibana/blob/master/x-pack/plugins/enterprise_search/common/types/index.ts#L30), at the bottom. Can we move these to the same file?

Reply:
I'm going to open up a new PR and i'll do it that wa.

Reply:
Which types specificall?

Reply:
Sorry about that; I selected the wrong lines. It's these, specifically:
```JavaScript
export interface IMetaPage {
  current: number;
  size: number;
  total_pages: number;
  total_results: number;
}

export interface IMeta {
  page: IMetaPage;
}
```

Reply:
That file will only hold type that are common to both app search and workplace search, correct?

Reply:
Correct